### ISSUE TITLE
Styles stored inside a GeoPackage are not found anymore by QGIS

### ISSUE THREAD

**Issue by brandtf** [qgis/QGIS#58467]
### What is the bug or the crash?

During the switch from QGIS-LTR 3.28 to QGIS-LTR 3.34, I noticed some serious problems in the recognition of layer styles inside a GPKG, which also persist in QGIS 3.38. I have a GPKG with multiple styles in table "layer_styles" (ca. 100). The GPKG serves as a template and the correct association of the correct styles and layers is normally done automatically by a script, as there are sometimes multiple saved styles for the same GPKG table (for points, lines and polygons).

![grafik](https://github.com/user-attachments/assets/b601e0e6-bf12-4609-bf38-afa34630173e)

Since the switch of the version, in some cases, for some layers (= GPKG tables), the function "layer.listStylesInDatabase()" remains empty, although there ARE styles saved inside the GPKG for the associated GPKG tables. The lack of found styles can be reproduced manually by a filled GPKG (see below). The styles are not visible in the GUI of QGIS 3.34 or 3.38 but are visible in QGIS 3.28.

GPKG-Download:
Link: https://dap.thueringen.de/s/sLmMdZki27qC5tZ
Password: Kh516C2Aj1yu

### Steps to reproduce the issue

I made an upload with a GPKG , as I was not be able to reproduce the issue with very "simple steps" (i.e. just make 3 layers and 3 styles). It only occures in big GPKGs with multiple styles, and even multiple styles for the same GPKG table.

1. Load the GPKG inside QGIS (Drag´n´Drop)
2. Load all Layers (as group)
3. Properties of Layer: ax_landwirtschaft
4. Style --> Load style --> From source database
5. You will see: No style associated with layer, only one other style in database (AX_Wohnplatz_P)
6. You can´t assign a style that is not shown (and even not found in PyQGIS).

![grafik](https://github.com/user-attachments/assets/c02a3d4b-4f7b-4ca0-9256-d3bad54178c4)

But:
1. Properties of Layer: ax_bauwerkimgewaesserbereich
2. Style --> Load style --> From source database
3. You will see multiple styles for the layer (different spellings for testing) and all the other styles.
4. That´s how it should be, But most of the other layers show no (other) styles.

![grafik](https://github.com/user-attachments/assets/a53fb3c7-c7d3-4a38-a763-9978affa1dcf)

In QGIS 3.28, all layer styles are found and displayed correctly, when you load the GPKG in the same way.

### Versions

QGIS-Version | 3.38.2-Grenoble | QGIS-Codeversion | 130c432394

Qt-Version | 5.15.13
Python-Version | 3.12.5
GDAL-Version | 3.9.2
PROJ-Version | 9.4.0
EPSG-Registraturdatenbankversion | v11.004 (2024-02-24)
GEOS-Version | 3.12.1-CAPI-1.18.1
SQLite-Version | 3.45.1
PDAL-Version | 2.6.3
PostgreSQL-Client-Version | 16.2
SpatiaLite-Version | 5.1.0
QWT-Version | 6.2.0
QScintilla2-Version | 2.14.1
BS-Version | Windows 11 Version 2009
  |   |   |  
Aktive Python-Erweiterungen
GeoBasis_Loader | 0.6
GeoCoding | 2.19
plugin_reloader | 0.9.4
db_manager | 0.1.20
grassprovider | 2.12.99
MetaSearch | 0.3.6
processing | 2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by agiudiceandrea** [comment]
On Windows the issue doesn't occur until 3.32.3 (GDAL/OGR 3.7.2) while it does occur since QGIS 3.34.0 (GDAL/OGR 3.7.2).
Hi @manisandro @elpaso , please have a look at this issue report about a regression in retrieving styles stored in a GPKG file.

**Comment by elpaso** [comment]
I cannot reproduce in current master.

**Comment by agiudiceandrea** [comment]
@elpaso, the issue occurs on Windows 10 even using QGIS 3.39.0-Master f90a7c04b1 (GDAL/OGR 3.10.0dev-b39ca5a8b5) form OSGeo4W.

![image](https://github.com/user-attachments/assets/c0d97160-b487-40cc-b8a2-9f633ead537b)

**Comment by elpaso** [comment]
@agiudiceandrea 

maybe my GDAL master fixed it?

![immagine](https://github.com/user-attachments/assets/d9e1f14e-2506-4a82-be7e-28c57c7b0c7d)

**Comment by agiudiceandrea** [comment]
@elpaso, I don't think so, since the issue does occur on my Windows 10 system even running the latest QGIS 3.39.0-Master 343709f48a with GDAL/OGR 3.10.0dev [526b28c086](https://github.com/OSGeo/gdal/commit/526b28c08612934cca2f8d1e8e2695eda730407f).

**Comment by nyalldawson** [comment]
The original download is no longer available

**Comment by agiudiceandrea** [comment]
This is the original file "example.gpkg" zipped in [example.zip](https://github.com/user-attachments/files/18762717/example.zip) (password: NaIIC80zZzUP).
The described issue still occurs using QGIS 3.40.3 on Windows 10 (OSGeo4W).

**Comment by agiudiceandrea** [comment]
@nyalldawson, have you managed to replicate the problem using the file attached in the previous comment?

**Comment by nyalldawson** [comment]
@agiudiceandrea I haven't looked -- it'll need to wait till the next funded bug fixing round
